core/src/main/scala/kafka/log/Log.scala

```diff
@@ -312,7 +320,7 @@ class Log(@volatile var dir: File,
       if(truncatedBytes > 0) {
         // we had an invalid message, delete all remaining log
         warn("Corruption found in segment %d of log %s, truncating to offset %d.".format(curr.baseOffset, name, curr.nextOffset))
-        unflushed.foreach(deleteSegment)
+        unflushed.foreach(deleteSegment(_))
```

Comment:
We can set the deleteEpoch flag to false here since we call clearLatest() after loading the log.